src/index.ts

```diff
@@ -1,19 +1,16 @@
 import {DependencyList, useEffect, useState} from 'react'
 
-export function useAsyncMemo<T>(factory: () => Promise<T> | undefined | null, deps: DependencyList, initial: T = undefined): T {
-  const [val, setVal] = useState<T>(initial)
+export function useAsyncMemo<T>(factory: (...args: any) => Promise<T>, deps?: DependencyList, initial?: T) {
+  let [val, setVal] = useState(initial)
+
   useEffect(() => {
-    let cancel = false
-    const promise = factory()
-    if (promise === undefined || promise === null) return
```

Comment:
There is some implicitly logic here: If the `factory` function returns `null` or `undefined`. The memoized value will stay unchanged.
If we remove this logic, there will be a break change. So we need to consider about it carefully. Do you think this is a necessary feature? Or maybe it is too obscure?

Reply:
This will fall away at the type checking stage. By the way, why should someone pass non-promise handler into hook with `async` in name